src/sage/schemes/elliptic_curves/ell_finite_field.py

```diff
@@ -2786,3 +2861,290 @@ def find_q(m, m4_fac, D):
                         yield Et
                     except ValueError:
                         pass
+
+def EllipticCurve_with_prime_order(N):
+    r"""
+    Given a prime number ``N``, find another prime number `p` and construct an
+    elliptic curve `E` defined over `\mathbb F_p` such that
+    `\#E(\mathbb F_p) = N`.
+
+    INPUT:
+
+    - ``N`` -- integer; the order for which we seek an elliptic curve. Must be a
+      prime number.
+
+    OUTPUT: an iterator of elliptic curves `E/\mathbb F_p` of order ``N``
+
+    .. NOTE::
+
+        Depending on the input, this function may run for a *very* long time.
+        This algorithm consists of multiple "search rounds" for a suitable
+        discriminant `D`. We expect this algorithm to terminate after a number
+        of rounds that is polynomial in `\log\log N`. In practice (cf. Section
+        5), this number is usually 1.
```

Comment:
```suggestion
        of rounds that is polynomial in `\log\log N`. In practice, this number
        is usually 1.
```

I don't like referring to a section of a paper without specifying which paper.

Alternatively:

```suggestion
        of rounds that is polynomial in `\log\log N`. In practice, this number
        is usually 1 (see Section 5 of [BS2007]_).
```

Reply:
Mmmh, since we don't do the "rounds" anymore, this comment is probably outdated or we could update the expectation using results from the paper.

Reply:
I will delete the note entirely, it's covered in the tests already